_posts/pl/2023-12-01-czy-wiesz-jak-uzywac-discriminated-union-w-jezyku-typescript.md

```diff
@@ -0,0 +1,53 @@
+---
+layout:    post
+title:     "Czy wiesz jak używać discriminated union w języku TypeScript?"
+date:      2023-12-01T08:00:00+01:00
+published: true
+didyouknow: true
+lang: pl
+author: kczechowski
+image: /assets/img/posts/2023-12-01-czy-wiesz-jak-uzywac-discriminated-union-w-jezyku-typescript/code.webp
+description: "Discriminated union  to połączenie takich typów, z których każdy posiada jedno wspólne pole, po którym możemy określić, z którym z nich mamy do czynienia..."
+tags:
+- typescript
+- discriminated union
+- literal type
+---
+Discriminated union (pol. *unia dyskryminacyjna*) to połączenie takich typów, z których każdy posiada jedno wspólne pole, po którym możemy określić, z którym z nich mamy do czynienia. Zgodnie z ogólnie przyjętą konwencją jest to pole `type`, które jest typu **Literal Type**, czyli jego typ opisuje się za pomocą stringa.
```

Comment:
propozycja uproszczenia i uczesania:
"...jest to pole `type`, czyli typ **Literal Type**, który opisuje się za pomocą stringa."